nodestream_akamai/edns/edns.py

```diff
@@ -1,57 +1,65 @@
+import asyncio
 import logging
-from ipaddress import IPv4Address, ip_address
 
 from nodestream.pipeline.extractors import Extractor
 
+from ..akamai_utils import addresses
 from ..akamai_utils.edns_client import AkamaiEdnsClient
 
+SUPPORTED_RECORD_TYPES = [
+    "A",
+    "AAAA",
+    "CNAME",
+    "NS",
+    "CAA",
+]
+
 
 class AkamaiEdnsExtractor(Extractor):
     def __init__(self, **akamai_client_kwargs) -> None:
         self.client = AkamaiEdnsClient(**akamai_client_kwargs)
         self.logger = logging.getLogger(self.__class__.__name__)
-        self.SUPPORTED_RECORD_TYPES = ["A", "AAAA", "CNAME", "NS", "CAA"]
-        self.ADDRESS_FORMAT_TO_NODE_TYPE = {
-            "ipv4": "Cidripv4",
-            "ipv6": "Cidripv6",
-            "endpoint": "Endpoint",
-        }
-
-    def _get_address_format(self, address):
+
+    def _extract_recordset(self, recordset, zone):
+        self.logger.debug(
+            "Extracting recordset %s/%s for zone %s",
+            recordset.get("name"),
+            recordset.get("type"),
+            zone,
+        )
+        recordset["key"] = f'{recordset["name"]}/{recordset["type"]}'
+        recordset["zone"] = zone
+        for record in recordset["rdata"]:
+            address_format = addresses.get_format(record)
+            node_type = address_format.node_type
+            if node_type:
+                recordset.get(node_type, []).append(record)
+        return recordset
+
+    async def _extract_zone(self, zone):
         try:
-            if type(ip_address(address)) is IPv4Address:
-                return "ipv4"
-            else:
-                return "ipv6"
-        except ValueError:
-            if " " in address:
-                return "Invalid"
-            else:
-                return "endpoint"
+            record_sets = self.client.list_recordsets(zone["zone"])
+        except Exception as e:
+            self.logger.error(
+                "Failed to list record sets for zone: %s",
+                zone["zone"],
+                exc_info=True,
+            )
+            raise e
```

Comment:
added the `raise` here because the code is going to explode anyway if you try to use `record_sets` after a failure making the http request.

This change slipped in due to initially having no permissions to the new endpoint paths and the error message confused me a bit.